### PR TITLE
15 remove all favorites

### LINKED ISSUE #12: User Story 15, Remove all Favorite from Favorites Page
As a visitor
When I have added pets to my favorites list
And I visit my favorites page ("/favorites")
I see a link to remove all favorited pets
When I click that link
I'm redirected back to the favorites page
I see the text saying that I have no favorited pets
And the favorites indicator returns to 0

### PR DESCRIPTION
**Pull request by jfangonilo**
## What functionality does this accomplish?
closes #12 
Add functionality to remove all favorites

## What did you struggle on to complete?
Using session.delete instead of reset_session

### Test Coverage Percentage: 100%
- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened):
- [x] New Tests added for new features

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code
- [ ] I have partially tested my code (please explain why):

## Helpful Resources:
* Resource link AND small description of info gathered/learned
https://stackoverflow.com/questions/2405635/how-to-empty-destroy-a-session-in-rails

## Please include a gif of how you feel about this branch:
3oz8xQQP4ahKiyuxHy

### CODE REVIEW

File: app/controllers/favorites_controller.rb [hunk 1]
```diff
@@ -20,4 +20,9 @@ def update
     redirect_to "/pets/#{pet.id}"
     flash[:notice] = "#{pet.name} has been added to your favorites!"
   end
+
+  def reset
+    session.delete(:favorites)
```
Comment (madelynrr) [inline app/controllers/favorites_controller.rb]: So does session.delete delete entire hash, or just the key/value pairs?

**Comment by madelynrr** [comment]
Looks good, once I finish delete one pet from favorites, need to decide on whether or not we want to make it more RESTful and put both in one action or not!

File: app/controllers/favorites_controller.rb (hunk 1 above)
Comment (jfangonilo) [reply app/controllers/favorites_controller.rb]: It deletes the whole session[:favorites] hash. nil is returned after it's called in line 25

**Comment by jfangonilo** [comment]
Agreed!

**Pull request merged**